core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -501,9 +521,8 @@ public TableMetadata replaceCurrentSnapshot(Snapshot snapshot) {
     return new Builder(this).setCurrentSnapshot(snapshot).build();
   }
 
-  public TableMetadata removeSnapshotsIf(Predicate<Snapshot> removeIf) {
```

Comment:
I think we still need to keep this method for backwards API compatibility, could you mark it as deprecated instead?

In the original PR I chose to skip removing snapshots with ref/branch, we can discuss what's the preferred behavior if we now decide to deprecate this method.